subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -1315,26 +1366,22 @@ int nrf_cloud_fota_mqtt_evt_handler(const struct mqtt_evt *evt)
 		case NRF_CLOUD_FOTA_VALIDATE_PASS:
 		case NRF_CLOUD_FOTA_VALIDATE_UNKNOWN:
 		case NRF_CLOUD_FOTA_VALIDATE_FAIL:
+		case NRF_CLOUD_FOTA_VALIDATE_ERROR:
 			save_validate_status(saved_job.id, saved_job.type,
 					NRF_CLOUD_FOTA_VALIDATE_DONE);
 			break;
 		case NRF_CLOUD_FOTA_VALIDATE_PENDING:
-		/* This event should cause reboot or modem-reinit.
-		 * Do not cleanup the job since a reboot should
-		 * occur or the user will call:
-		 * nrf_cloud_modem_fota_completed() or nrf_cloud_fota_pending_job_validate()
-		 * The validated job status will be sent after the reboot or upon
-		 * reconnection to nRF Cloud (see nrf_cloud_fota_endpoint_set_and_report())
-		 */
-			send_event(NRF_CLOUD_FOTA_EVT_DONE, &current_fota);
+			send_fota_done_event();
 			break;
 		default:
 			break;
 		}
 		break;
 	}
-	case MQTT_EVT_CONNACK:
 	case MQTT_EVT_DISCONNECT:
+		nrf_cloud_fota_endpoint_clear();
+		break;
```

Comment:
I think this is intentional, but just to be clear, the intent here is to prevent `nct_mqtt_evt_handler` from responding to disconnects during a FOTA update, right?

If so, what is the reason -- and maybe there should be a comment explaining that?

Reply:
no, the purpose is to clear the mqtt endpoint information in this fota module so it won't try to send mqtt messages.

Reply:
Oops, my review comment was pretty vague, sorry,

I meant the fact that this now returns 0 instead of 1 in the case of MQTT_EVT_DISCONNECT

Reply:
ah,  yes.  it should still return 1!